plugins/source/aws/resources/services/rds/engine_versions_fetch.go

```diff
@@ -24,3 +27,11 @@ func fetchRdsEngineVersions(ctx context.Context, meta schema.ClientMeta, _ *sche
 	}
 	return nil
 }
+func calculateUniqueHash(_ context.Context, _ schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	f := resource.Item.(types.DBEngineVersion)
+	hash, err := hashstructure.Hash(f, hashstructure.FormatV2, nil)
```

Comment:
does result differ if we pass just `resource.Item`?

Reply:
I will look into it

Reply:
My concern with this is that if a new field gets added to the struct, it will be added to the hash too.
For example a new `UpdateTime` field that will break the uniqueness.
On the other hand it could be considered a feature if the field does contribute to uniqueness 🙃 
Another use case is that AWS removes one field and adds another, then the hash changes.

Maybe we should hard code the list of fields that are a part of the hash and error if it changes?

Reply:
I think it's OK to leave full hash, as this is a despperate measure aimed solely at providing uniqueness to the resource PK.

Reply:
> does result differ if we pass just `resource.Item`?

Hash value is the same... So removed the casting

Reply:
I agree with @candiduslynx... Because this is fully defined by AWS and there is absolutely no guidance on uniqueness I think anything else that we do will leave us in a more brittle position

Reply:
I think I didn't explain my suggestion properly.
It's very close to the existing solution.
I'm suggesting we have a hard coded list of all **current** fields of `DBEngineVersion` and use those as a hash.
Then if while creating the hash we detect a new/removed field we fail (our tests should catch this failure).
That way the current logic stays the same, but if even AWS changes the structure of `DBEngineVersion` we'll know.

Reply:
And **do** we want to know?

Reply:
I can see 2 cases we'd like to know:
1. AWS adds a new field that can serve as a PK so we can drop this fix
2. AWS adds a new field that breaks uniqueness (like a timestamp) meaning this fix stops working

Regardless my comment is non blocking for the PR as the changes improve on the current state.

Reply:
Added an issue to track this... https://github.com/cloudquery/cloudquery/issues/7293